cloudcix_primitives/rbd_lxd.py

```diff
@@ -1,26 +1,396 @@
+    
+    if status is False:
+        return status, result, msg
+
+    return True, result, f'1000: {messages[1000]}'
+
+
+def scrub(
+    endpoint_url: str,
+    pool_name: str,
+    volume_name: str,
+    verify_lxd_certs: bool = True,
+) -> Tuple[bool, str]:
+    """
+    description: Scrub (delete) a RadOS Block Device via LXD host.
+    
+    parameters:
+        endpoint_url:
+            description: The endpoint URL for the LXD Host.
+            type: string
+            required: true
+        pool_name:
+            description: Name of the storage pool.
+            type: string
+            required: true
+        volume_name:
+            description: Name of the volume to delete.
+            type: string
+            required: true
+        verify_lxd_certs:
+            description: Boolean to verify LXD certs.
+            type: boolean
+            required: false
+            
+    return:
+        description: A tuple with a boolean flag indicating success or failure, and a message.
+        type: tuple
+    """
+    # Define message
+    messages = {
+        1000: f'Successfully scrubbed RadOS Block Device {volume_name} from pool {pool_name} on {endpoint_url}',
+        3001: f'Failed to connect to {endpoint_url} for storage pool operations',
+        3002: f'Failed to get storage pool {pool_name} on {endpoint_url}',
+        3003: f'Failed to delete volume {volume_name} from pool {pool_name}',
+        3004: f'Volume {volume_name} not found in pool {pool_name}',
+    }
+
+    def run_host(endpoint_url, prefix, successful_payloads):
+        rcc = LXDCommsWrapper(comms_lxd, endpoint_url, verify_lxd_certs)
+        fmt = HostErrorFormatter(
+            endpoint_url,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+        
+        # STEP 1: Confirm the Ceph storage pool exists
+        ret = rcc.run(cli='storage_pools.get', name=pool_name)
+        
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+2}: {messages[prefix+2]}"), fmt.successful_payloads
+
+        pool_obj = ret["payload_message"]
+        fmt.add_successful('storage_pools.get', ret)
+        
+        # STEP 2: Check if the pool has the required methods
+        if not hasattr(pool_obj, 'volumes') or not hasattr(pool_obj.volumes, 'get'):
+            return False, f"{prefix+3}: {messages[prefix+3]} - Missing required methods on pool object", fmt.successful_payloads
+        
+        # STEP 3: Try to get the volume first to ensure it exists
+        try:
+            volume = pool_obj.volumes.get("custom", volume_name)
+            
+            if not volume:
+                return False, f"{prefix+4}: {messages[prefix+4]}", fmt.successful_payloads
```

Comment:
As this is a scrub request, the volume not existing indicates it has been deleted already so True can be returned